src/go/pt-k8s-debug-collector/dumper/dumper.go

```diff
@@ -0,0 +1,355 @@
+	}
+
+	return outb.Bytes(), nil
+}
+
+func (d *Dumper) getResource(name, namespace string, tw *tar.Writer) error {
+	location := d.location
+	args := []string{"get", name, "-o", "yaml"}
+	if len(namespace) > 0 {
+		args = append(args, "--namespace", namespace)
+		location = d.location + "/" + namespace
+	}
+	location += "/" + name + ".yaml"
+	output, err := d.runCmd(args...)
+	if err != nil {
+		d.logError(err.Error(), args...)
+		log.Printf("Error: get resource %s in namespace %s: %v", name, namespace, err)
+		return addToArchive(location, d.mode, []byte(err.Error()), tw)
+	}
+
+	return addToArchive(location, d.mode, output, tw)
+}
+
+func (d *Dumper) logError(err string, args ...string) {
+	d.errors += d.cmd + " " + strings.Join(args, " ") + ": " + err + "\n"
+}
+
+func addToArchive(location string, mode int64, content []byte, tw *tar.Writer) error {
+	hdr := &tar.Header{
+		Name: location,
+		Mode: mode,
+		Size: int64(len(content)),
+	}
+	if err := tw.WriteHeader(hdr); err != nil {
+		return errors.Wrapf(err, "write header to %s", location)
+	}
+	if _, err := tw.Write(content); err != nil {
+		return errors.Wrapf(err, "write content to %s", location)
+	}
+
+	return nil
+}
+
+type crSecrets struct {
+	Spec struct {
+		SecretName string `json:"secretsName,omitempty"`
+		Secrets    struct {
+			Users string `json:"users,omitempty"`
+		} `json:"secrets,omitempty"`
+	} `json:"spec"`
+}
+
+func (d *Dumper) getPTSummury(resource, podName, crName string, tw *tar.Writer) ([]byte, error) {
+	var (
+		summCmdName string
+		ports       string
+		summCmdArgs []string
+	)
+
+	switch resource {
+	case "pxc":
+		cr, err := d.getCR("pxc/" + crName)
+		if err != nil {
+			return nil, errors.Wrap(err, "get cr")
+		}
+		pass, err := d.getDataFromSecret(cr.Spec.SecretName, "root")
+		if err != nil {
+			return nil, errors.Wrap(err, "get password from pxc users secret")
+		}
+		ports = "3306:3306"
+		summCmdName = "pt-mysql-summary"
+		summCmdArgs = []string{"--host=127.0.0.1", "--port=3306", "--user=root", "--password=" + string(pass)}
+	case "psmdb":
+		cr, err := d.getCR("psmdb/" + crName)
+		if err != nil {
+			return nil, errors.Wrap(err, "get cr")
+		}
+		pass, err := d.getDataFromSecret(cr.Spec.Secrets.Users, "MONGODB_CLUSTER_ADMIN_PASSWORD")
+		if err != nil {
+			return nil, errors.Wrap(err, "get password from psmdb users secret")
+		}
+		ports = "27017:27017"
+		summCmdName = "pt-mongodb-summary"
+		summCmdArgs = []string{"--username=clusterAdmin", "--password=" + pass, "--authenticationDatabase=admin", "127.0.0.1:27017"}
+	}
+
+	cmdPortFwd := exec.Command(d.cmd, "port-forward", "pod/"+podName, ports)
+	go func() {
+		err := cmdPortFwd.Run()
+		if err != nil {
+			d.logError(err.Error(), "port-forward")
+		}
+	}()
+	defer func() {
+		err := cmdPortFwd.Process.Kill()
+		if err != nil {
+			d.logError(err.Error(), "kill port-forward")
+		}
+	}()
+
+	time.Sleep(3 * time.Second) // wait for port-forward command
```

Comment:
Why not exec.Start() and exec.Wait()?

Reply:
Because port-forward function will work endless